### ISSUE TITLE
Having similar properties breaks `@required_fields`

### ISSUE THREAD

**Issue by catgirlinspace** [MrThearMan/graphene-django-query-optimizer#65]
### I have read the docs thoroughly before making this bug report.

- [X] Yes

### I have read through other open issues, and my issue is not a duplicate.

- [X] Yes

### What version of the library you are using?

0.1.11

### Which python version are you using?

3.12

### What operating system are you on?

Mac

### Description

Hi, I have a project I'm trying to use this library in, but found a bug where having similar properties on a model breaks `@required_fields`. Specifically, I have this in my GraphQL resolver thing:
```python
@required_fields('nameplate_badge_1')
    def resolve_splashtag(parent, info):
        return parent.splashtag
```
(parent.splashtag uses nameplate_badge_1 and some other foreign keys, but for simplicity I'm only using `nameplate_badge_1` here)

and my model has these fields
```python
 name = models.CharField(max_length=50)
    name_id = models.CharField(max_length=10, blank=True, null=True)
    ...
    nameplate_background = models.ForeignKey('splatnet_assets.NameplateBackground', on_delete=models.PROTECT)
    nameplate_badge_1 = models.ForeignKey('splatnet_assets.NameplateBadge', on_delete=models.PROTECT, null=True,
                                          related_name='+')
    nameplate_badge_2 = models.ForeignKey('splatnet_assets.NameplateBadge', on_delete=models.PROTECT, null=True,
                                          related_name='+')
    nameplate_badge_3 = models.ForeignKey('splatnet_assets.NameplateBadge', on_delete=models.PROTECT, null=True,
                                          related_name='+')
```
It seems like the loop [here](https://github.com/MrThearMan/graphene-django-query-optimizer/blob/e8958708426747ab7782c92bf9efacda69b940e5/query_optimizer/optimizer.py#L232) breaks when the `model_field` is name because `nameplate_badge_1` does start with `name`? I'm not sure how to resolve this really...

**Comment by MrThearMan** [comment]
Hi, can you include an error traceback as a comment? Having a bit more information on that would help me replicate the issue.

**Comment by catgirlinspace** [comment]
Doesn't really seem to have a traceback :( I just get this from GraphQL
```json
{
  "errors": [
    {
      "message": "No related model, but hint seems like has one: 'nameplate_badge_1'",
      "locations": [
        {
          "line": 5,
          "column": 9
        }
      ],
      "path": [
        "users",
        "edges",
        0,
        "node",
        "battles"
      ]
    }
  ],
  "data": {
    "users": {
      "edges": [
        {
          "node": null
        }
      ]
    }
  }
}
```

**Comment by MrThearMan** [comment]
Ok, this one is a silly one, the [startswith check](https://github.com/MrThearMan/graphene-django-query-optimizer/blob/e8958708426747ab7782c92bf9efacda69b940e5/query_optimizer/optimizer.py#L241) is meant to check if the field might be on a related entity (i.e. `related__field`), but it recognizes things like "name" as being relations. Easy fix, added to the main branch.

I'm currently working on #61 and #62 and will include those in the next release as well, so this will need to wait for that. Should get those one this week.

**Comment by MrThearMan** [comment]
Released in [0.2.0](https://github.com/MrThearMan/graphene-django-query-optimizer/releases/tag/v0.2.0)
